ui/src/app/settings/components/project-details/project-details.tsx

```diff
@@ -473,225 +435,627 @@ export class ProjectDetails extends React.Component<RouteComponentProps<{name: s
-                                    <div className='columns small-3'>{res.group}</div>
-                                    <div className='columns small-6'>{res.kind}</div>
+                            {(formApi.values.spec.clusterResourceWhitelist || []).map((_: Project, i: number) => (
+                                <div className='row white-box__details-row' key={i}>
+                                    <div className='columns small-4'>
+                                        <FormField
+                                            formApi={formApi}
+                                            field={`spec.clusterResourceWhitelist[${i}].kind`}
+                                            component={AutocompleteField}
+                                            componentProps={{items: ResourceKinds}}
+                                        />
+                                    </div>
+                                    <div className='columns small-8'>
+                                        <FormField
+                                            formApi={formApi}
+                                            field={`spec.clusterResourceWhitelist[${i}].group`}
+                                            component={AutocompleteField}
+                                            componentProps={{items: Groups}}
+                                        />
+                                    </div>
+                                    <i
+                                        className='fa fa-times'
+                                        onClick={() => formApi.setValue('spec.clusterResourceWhitelist', removeEl(formApi.values.spec.clusterResourceWhitelist, i))}
+                                    />
                                 </div>
+                            ))}
+                            <button
+                                className='argo-button argo-button--short'
+                                onClick={() =>
+                                    formApi.setValue(
+                                        'spec.clusterResourceWhitelist',
+                                        (formApi.values.spec.clusterResourceWhitelist || []).concat({
+                                            group: '*',
+                                            kind: '*'
+                                        })
+                                    )
+                                }>
+                                ADD RESOURCE
+                            </button>
+                        </React.Fragment>
+                    )}
+                    items={[]}
+                />
+                <EditablePanel
+                    save={item => this.saveProject(item)}
+                    values={proj}
+                    title={<React.Fragment>CLUSTER RESOURCE DENY LIST {helpTip('Cluster-scoped K8s API Groups and Kinds which are not permitted to be deployed')}</React.Fragment>}
```

Comment:
It feels like this change from `blacklist`/`whitelist` to `deny list`/`allow list` should have been called out in a commit message / release note...